File: solutions/typescript/bob/1/bob.ts
const isQuestion = (text: string): boolean => {
  return text[text.length - 1] === '?'
}

const isYelling = (text: string): boolean =>{
  return text === text.toUpperCase() && text !== text.toLowerCase();
}

const isOnlyWhitespace = (text: string) => {
    return text.trim() === '';
}


export function hey(message: string): string {

  if(isYelling(message)) return isQuestion(message) ? "Calm down, I know what I'm doing!" :  "Whoa, chill out!"

  if(isQuestion(message.trim())) return 'Sure.'
  
  if(isOnlyWhitespace(message)) return "Fine. Be that way!"

  return "Whatever."
  
}

File: solutions/kotlin/bob/1/src/main/kotlin/Bob.kt
object Bob {
    fun hey(input: String): String {
        val trimmed = input.trim()

        if (trimmed.isBlank()) return "Fine. Be that way!"

        val isQuestion: Boolean = trimmed.endsWith('?')
        val hasLetter: Boolean = input.any { c -> c in 'a'..'z'  || c in 'A'..'Z' }
        val isYelling: Boolean = hasLetter && trimmed == trimmed.uppercase()

        return when {
            isQuestion && isYelling -> "Calm down, I know what I'm doing!"
            isQuestion -> "Sure."
            isYelling -> "Whoa, chill out!"
            else -> "Whatever."
        }

    }
}
